components/Tabs/src/MDCTabBar.h

```diff
@@ -152,3 +172,28 @@ IB_DESIGNABLE
 - (void)tabBar:(nonnull MDCTabBar *)tabBar didSelectItem:(nonnull UITabBarItem *)item;
 
 @end
+
+/**
+ Fixed configuration options for tab bars.
+
+ The tab bar copies the configuration passed at initialization, so all configuration changes must
+ be made before creating the tab bar. 
+*/
+@interface MDCTabBarConfiguration : NSObject <NSCopying>
```

Comment:
I really like this idea!

My only concern would be that it makes the default values for affected properties weird. For example, the default alignment for top tabs should be Leading but for bottom tabs should be Justified. Similar deal for the other two properties that are public on the configuration. 

What would [[MDCTabBar alloc] init].alignment return? What if a delegate is set?
A. Specifically Leading or Justified, chosen arbitrarily or based on whatever the delegate returns.
B. A new sentinel alignment case Default that specifies it's whatever is the default value for the context. This approach works for enums and objects, but would require changing BOOL properties into an enum.
C. Explicitly undefined - querying the alignment before setting it is not correct use of the API. Prone to misuse, but clearly simplest to implement.